src/lib.rs

```diff
@@ -1,3 +1,47 @@
+
+pub trait Consensus<B: AsRef<u8>> {
+    // Finalised transaction callback function type.
+    //
+    // The callback function must accept only one parameter of type B; and
```

Comment:
Why can't we impose those conditions in the code? What's wrong with defining `Callback` as the example suggest?

Reply:
If I put type inside trait it's considered as associated type and compiler gives an error saying something like default value for associated type is not supported.
Should I put definition of that type outside of the trait then?

Reply:
I have made it that way: https://github.com/Fantom-foundation/libconsensus/blob/bd8b645a4d1737197b85a546ed3e72157b48daaa/src/lib.rs#L23